concrete/dense_directed_matrix.go

```diff
@@ -112,8 +115,16 @@ func (g *DirectedDenseGraph) HasEdgeFromTo(u, v graph.Node) bool {
 	return uid != vid && !isSame(g.mat.At(uid, vid), g.absent)
 }
 
-func (g *DirectedDenseGraph) Weight(e graph.Edge) float64 {
-	return g.mat.At(e.From().ID(), e.To().ID())
+func (g *DirectedDenseGraph) Weight(x, y graph.Node) (w float64, ok bool) {
+	xid := x.ID()
+	yid := y.ID()
+	if xid == yid {
+		return 0, true
```

Comment:
Sorry for unnecessary questions, I got to the Weighter interface in the meantime.

What other reasonable values are there for absent apart from Inf(1)? Accepting that any number, then 1/absent feels a bit arbitrary.

At any case, for completeness, shouldn't there be an (unlikely, but possible) subcase that `xid == yid && !g.has(xid)` ?